File: Backend/src/main/java/lk/dilan/project/backend/entity/Events.java
package lk.dilan.project.backend.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Document(collection = "events")
@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
public class Events {
    @Id
    private String id;
    private String userID;/*FK from user table*/
    private String title;
    private String description;
    private LocalDate date;
    private String location;
    private int totalSeats;
    private int availableSeats;
    private double price;
    private String eventImageBase64;
    private String imageName;
}

File: FrontEnd/src/types/Events.ts
export interface EventForm {
    title: string;
    userId:string
    description: string;
    date: string;
    location: string;
    totalSeats: number;
    price: number;
    eventImageBase64: string | null;
    eventImageFileName: string | null;
}


export interface EventPost {
    id: string;
    title: string;
    description: string;
    date: string;
    location: string;
    totalSeats: number;
    price: number;
    eventImageBase64: string | null;
    eventImageFileName: string | null;
}

export interface UserEvent {
    id: string;
    title: string;
    description: string;
    date: string;
    location: string;
    totalSeats: number;
    price: number;
    eventImageBase64: string | null;
    eventImageFileName: string | null;
    isBought: boolean;
}

export interface PaymentData {
    userId: string | null;
    eventId: string;
    amount: number;
    method: string; // e.g., "CARD"
}